src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Bit.App.Resources;
+using Bit.App.Utilities;
+using Bit.Core.Abstractions;
+using Bit.Core.Utilities;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.Forms;
+
+namespace Bit.App.Pages
+{
+    public class LoginPasswordlessViewModel : BaseViewModel
+    {
+        private IAuthService _authService;
+        private IPlatformUtilsService _platformUtilsService;
+        private ILogger _logger;
+        private string _logInAttempByLabel;
+        private string _deviceType;
+        private FormattedString _fingerprintPhraseFormatted;
+        private string _fingerprintPhrase;
+        private string _email;
+        private string _timeOfRequest;
+        private DateTime _requestDate;
+        private string _nearLocation;
+        private string _ipAddress;
+
+        public LoginPasswordlessViewModel()
+        {
+            _authService = ServiceContainer.Resolve<IAuthService>("authService");
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+            _logger = ServiceContainer.Resolve<ILogger>("logger");
+
+            PageTitle = AppResources.LogInRequested;
+
+            AcceptRequestCommand = new AsyncCommand(AcceptRequestAsync,
+                onException: ex => _logger.Exception(ex),
+                allowsMultipleExecutions: false);
+            RejectRequestCommand = new AsyncCommand(RejectRequestAsync,
+                onException: ex => _logger.Exception(ex),
+                allowsMultipleExecutions: false);
+        }
+
+        public ICommand AcceptRequestCommand { get; }
+
+        public ICommand RejectRequestCommand { get; }
+
+        public string Origin { get; set; }
+
+        public string Email
+        {
+            get => _email;
+            set
+            {
+                LogInAttempByLabel = string.Format(AppResources.LogInAttemptByOn, value, Origin);
+                SetProperty(ref _email, value);
+            }
+        }
+
+        public string FingerprintPhrase
+        {
+            get => _fingerprintPhrase;
+            set
+            {
+                FingerprintPhraseFormatted = CreateFingerprintPhrase(value);
+                SetProperty(ref _fingerprintPhrase, value);
+            }
+        }
+
+        public FormattedString FingerprintPhraseFormatted
+        {
+            get => _fingerprintPhraseFormatted;
+            set => SetProperty(ref _fingerprintPhraseFormatted, value);
+        }
+
+        public string LogInAttempByLabel
+        {
+            get => _logInAttempByLabel;
+            set => SetProperty(ref _logInAttempByLabel, value);
+        }
```

Comment:
Instead of having a full property here and assign it on the `Email`, this should be a `get` only property which returns the formatted string.